feat(firebaseai): add `responseJsonSchema` to `GenerationConfig`

gspencergoog opened the pull request:
## Description

This adds the `responseJsonSchema` field to the GenerationConfig class, which enables more JSON schema features like `$def` and `$ref` along with other advanced JSON schema features that Gemini 2.5 supports.

See [the Gemini docs](https://ai.google.dev/gemini-api/docs/structured-output#json-schema) for more information about what is supported.

## Related Issues

- Fixes https://github.com/firebase/flutterfire/issues/17551

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

cynthiajoan commented on packages/firebase_ai/firebase_ai/CHANGELOG.md:
We don't need to update this file manually, it would auto generated through the release process with corresponding PR title

gspencergoog replied:
Ahh, OK, sorrry it's my first time in this repo.  Reverted.

cynthiajoan commented on packages/firebase_ai/firebase_ai/lib/src/api.dart:
```diff
@@ -1020,6 +1022,17 @@ final class GenerationConfig extends BaseGenerationConfig {
+  /// class supports.  See [the Gemini
+  /// documentation](https://ai.google.dev/gemini-api/docs/structured-output#json-schema)
+  /// about the limitations of this feature.
+  final String? responseJsonSchema;
```
what's the purpose of using a String as the type, instead of for json object we usually can do Map<String, Object?>?

gspencergoog replied:
I was wrong about it wanting a string: it expects a JS object anyhow.

gspencergoog commented on packages/firebase_ai/firebase_ai/pubspec.yaml:
Reverted

gspencergoog commented:
> Thanks for the PR, would like to check if it has been tested under both vertexAI and googleAI backend.

Yes, I've tested it on both backends.

gspencergoog commented:
Okay, @cynthiajoan I added an example here that uses `$def` and `$anchor`, which wasn't possible before this change.

I also formatted the output a little better in the "chat" of the example, and updated the regular schema example to match.

I also bumped the Gemini model versions in the example's main, since they were pretty old, but I'm happy to revert that if you wanted them to stay that way.
